base/linalg/triangular.jl

```diff
@@ -1474,91 +1660,45 @@ function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
 
     # Compute accurate superdiagonal of T
     p = 1 / 2^s
+    A = complex(A)
+    blockpower!(A0, A, p)
+    return A,m,s
+
+end
+
+# Compute accurate diagonal and superdiagonal of A = A0^p
+function blockpower!(A0::UpperTriangular, A::UpperTriangular, p)
```

Comment:
By convention, `xyz!(y, a, b)` methods are expected to modify the first argument. See e.g. `A_mul_B!(y, A, B)`. It's better to swap the order of `A` and `A0`.